File: src/test/kotlin/io/techery/mappery/test/converter/ABConverter.java
package io.techery.mappery.test.converter;

import org.jetbrains.annotations.NotNull;

import io.techery.mappery.Converter;
import io.techery.mappery.MapperyContext;

public class ABConverter implements Converter<ABConverter.A, String> {
    @Override
    public String convert(@NotNull MapperyContext context, A source) {
        return source.toString();
    }

    public static class A {
    }

    public static class B extends A {
    }
}

File: src/test/kotlin/io/techery/mappery/test/converter/ABConverter.kt
package io.techery.mappery.test.converter

import io.techery.mappery.Converter
import io.techery.mappery.MapperyContext

class ABConverter : Converter<ABConverter.A, String> {

    override fun convert(context: MapperyContext, source: A): String {
        return source.toString()
    }

    open class A

    class B : A()
}
